### PR TITLE
Improve redis queue error handling

### PR DESCRIPTION
**Pull request by kiootic**
ref DEV-1481, DEV-1483

~NOTE: there's a breaking change with respect to the user import API status field, not sure if want to add a new field to indicate success/failure instead.~

### CODE REVIEW

File: docs/specs/user-import.md [hunk 1]
```diff
@@ -153,7 +153,7 @@ You will receive a response similar to the following when the import completed (
 {
   "id": "some_opaque_string",
   "created_at": "2024-01-01T00:00:00.000Z"
-  "status": "completed",
+  "status": "succeeded",
```
Comment (louischan-oursky) [inline docs/specs/user-import.md:156]: Do not change this.

File: pkg/lib/infra/redisqueue/task.go [hunk 2]
```diff
@@ -10,7 +10,8 @@ type TaskStatus string
 
 const (
 	TaskStatusPending   TaskStatus = "pending"
-	TaskStatusCompleted TaskStatus = "completed"
+	TaskStatusFailed    TaskStatus = "failed"
+	TaskStatusSucceeded TaskStatus = "succeeded"
```
Comment (louischan-oursky) [inline pkg/lib/infra/redisqueue/task.go:14]: Keep completed.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: pkg/redisqueue/consumer.go [hunk 3]
```diff
@@ -168,6 +169,21 @@ func (c *Consumer) dequeue(ctx context.Context) (*redisqueue.Task, *deps.AppProv
...
 }
 
+func (c *Consumer) process(
+	ctx context.Context,
+	task *redisqueue.Task,
+	appProvider *deps.AppProvider,
+) (result json.RawMessage, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("task panic: %s", r)
```
Comment (louischan-oursky) [inline pkg/redisqueue/consumer.go:179]: Use panicutil.MakeError to turn the return value of recover() into an error.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: pkg/redisqueue/consumer.go (hunk 3 above)
Comment (kiootic) [reply pkg/redisqueue/consumer.go]: Fixed.

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**